contracts/stake/src/wasm.rs

```diff
@@ -44,49 +45,51 @@ impl StakeContract {
         let counter = stake.counter();
 
         stake.increment_counter();
-        stake.insert_amount(value, dusk_abi::block_height());
+        stake.insert_amount(value, rusk_abi::block_height());
 
         // required since we're holding a mutable reference to a stake and
         // `dusk_abi::transact_raw` requires a mutable reference to the state
         drop(stake);
 
         // verify the signature is over the correct message
         let message = Self::stake_sign_message(counter, value);
-        let valid_signature = rusk_abi::verify_bls_sign(signature, pk, message);
+        let valid_signature =
+            rusk_abi::verify_bls(message, APK::from(&pk), signature);
```

Comment:
Perhaps we should accept  `APK`s as arguments instead of just `PublicKey`s. They implement the necessary `rkyv` traits as well.